src/commands/commerce/extension/register.ts

```diff
@@ -0,0 +1,144 @@
+        );
+        if (epnVal.result.totalSize === 0) {
+            throw new SfdxError(msgs.getMessage('extension.register.errEPN'));
+        }
+        if (storeRegisteredName === undefined) {
+            throw new SfdxError(msgs.getMessage('extension.register.undefinedName'));
+        }
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const results = forceDataRecordCreate(
+            'RegisteredExternalService',
+            `DeveloperName=${storeRegisteredName} MasterLabel=${storeRegisteredName} ExtensionPointName=${storeEPN} ExternalServiceProviderId=${apexClassId} ExternalServiceProviderType='Extension'`,
+            storeUserName
+        );
+        // Validates/checks for already existing unique name?
+        if (results instanceof SfdxError) {
+            throw new SfdxError(msgs.getMessage('extension.register.error', [storeRegisteredName, results.message]));
+        }
+        // JSON response of inserted record
+        return this.getInsertedRecord(storeRegisteredName, storeApexClass, storeUserName);
+    }
+
+    // Queries for apexclassId from existing apexClass table
+    private getApexClass(storeApexClass: string, storeUserName: string): string {
+        let apexClassId: string;
+        try {
+            apexClassId = forceDataSoql(
+                `SELECT Id FROM ApexClass WHERE Name='${storeApexClass}' LIMIT 1`,
+                storeUserName
+            ).result.records[0].Id;
+        } catch (error) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            const errorMsg = msgs.getMessage('extension.register.errApexClass', [storeApexClass, error.message]);
+            throw new SfdxError(errorMsg);
+        }
+        return apexClassId;
+    }
+    // returns entire record from RegisteredExternalService in JSON format
+    private getInsertedRecord(storeRegisteredName: string, storeApexClass: string, storeUserName: string): string {
+        const RegisteredTable = forceDataSoql(
+            `SELECT Id,ConfigUrl,DeveloperName,DocumentationUrl,ExtensionPointName,ExternalServiceProviderId,ExternalServiceProviderType,Language,MasterLabel,NamespacePrefix from RegisteredExternalService WHERE DeveloperName='${storeRegisteredName}'`,
+            storeUserName
+        );
+
+        for (const element of RegisteredTable.result.records) {
+            const finalTable = {
+                UniqueExtensionId: element['Id'],
+                ApexClassId: element['ExternalServiceProviderId'] as string,
+                RegisteredExtensionName: element['DeveloperName'] as string,
+                ExtensionPointName: element['ExtensionPointName'] as string,
+                ExternalServiceProviderType: element['ExternalServiceProviderType'] as string,
+            };
+            const returnResult = `${JSON.stringify(finalTable, null, 4)};\n`;
+            this.ux.log(`Registered External Service Table = ${returnResult}`);
```

Comment:
can we have the output the json object. I think having "Registered External Service Table = "  makes its sound like this is the entire table
```
Registered External Service Table = {
    "UniqueExtensionId": "1uuxx00000000pdAAA",
    "ApexClassId": "01pxx0000004UkGAAU",
    "RegisteredExtensionName": "JinsuTicket",
    "ExtensionPointName": "CommerceDx_TicketProcessing",
    "ExternalServiceProviderType": "Extension"
};
```
just
```
 {
    "UniqueExtensionId": "1uuxx00000000pdAAA",
    "ApexClassId": "01pxx0000004UkGAAU",
    "RegisteredExtensionName": "JinsuTicket",
    "ExtensionPointName": "CommerceDx_TicketProcessing",
    "ExternalServiceProviderType": "Extension"
}